<!DOCTYPE html>
<html lang="en">

  <!-- Head -->
  <head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">    <!-- Metadata, OpenGraph and Schema.org -->
    

    <!-- Standard metadata -->
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Abhilesh  Dhawanjewar</title>
    <meta name="author" content="Abhilesh  Dhawanjewar">
    <meta name="description" content="A simple, whitespace theme for academics. Based on [*folio](https://github.com/bogoli/-folio) design.
">
    <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website">


    <!-- Bootstrap & MDB -->
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous">

    <!-- Bootstrap Table -->
    <link defer rel="stylesheet" href="https://unpkg.com/bootstrap-table@1.21.4/dist/bootstrap-table.min.css">

    <!-- Fonts & Icons -->
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous">
    <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons">

    <!-- Code Syntax Highlighting -->
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="" id="highlight_theme_light">

    

    <!-- Styles -->
    
    <link rel="shortcut icon" href="/assets/img/favicon-32x32.png">
    
    <link rel="stylesheet" href="/assets/css/main.css">
    <link rel="canonical" href="https://abhilesh.github.io/">

    <!-- Dark Mode -->
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/native.css" media="none" id="highlight_theme_dark">

    <script src="/assets/js/theme.js"></script>
    <script src="/assets/js/dark_mode.js"></script>
    

  </head>

  <!-- Body -->
  <body class="fixed-top-nav sticky-bottom-footer">

    <!-- Header -->
    <header>

      <!-- Nav Bar -->
      <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top">
        <div class="container">
          <!-- Social Icons -->
          <div class="navbar-brand social">
            <a href="mailto:%61.%64%68%61%77%61%6E%6A%65%77%61%72@%75%63%6C.%61%63.%75%6B" title="email"><i class="fas fa-envelope"></i></a>
            <a href="https://scholar.google.com/citations?user=KmALIboAAAAJ" title="Google Scholar" rel="external nofollow noopener" target="_blank"><i class="ai ai-google-scholar"></i></a>
            <a href="https://github.com/abhilesh" title="GitHub" rel="external nofollow noopener" target="_blank"><i class="fab fa-github"></i></a>
            <a href="https://www.linkedin.com/in/abhilesh-dhawanjewar" title="LinkedIn" rel="external nofollow noopener" target="_blank"><i class="fab fa-linkedin"></i></a>
            

          </div>
          
          <!-- Navbar Toggle -->
          <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar top-bar"></span>
            <span class="icon-bar middle-bar"></span>
            <span class="icon-bar bottom-bar"></span>
          </button>

          <div class="collapse navbar-collapse text-right" id="navbarNav">
            <ul class="navbar-nav ml-auto flex-nowrap">

              <!-- About -->
              <li class="nav-item active">
                <a class="nav-link" href="/">About<span class="sr-only">(current)</span></a>
              </li>
              

              <!-- Other pages -->
              <li class="nav-item ">
                <a class="nav-link" href="/publications/">Publications</a>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/cv/">CV</a>
              </li>

              <!-- Toogle theme mode -->
              <li class="toggle-container">
                <button id="light-toggle" title="Change theme">
                  <i class="fas fa-moon"></i>
                  <i class="fas fa-sun"></i>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </nav>

      <!-- Scrolling Progress Bar -->
      <progress id="progress" value="0">
        <div class="progress-container">
          <span class="progress-bar"></span>
        </div>
      </progress>
    </header>


    <!-- Content -->
    <div class="container mt-5">
      
        <!-- about.html -->
      <div class="post">
        <header class="post-header">
          <h1 class="post-title">
           Abhilesh  <span class="font-weight-bold">Dhawanjewar</span>
          </h1>
          <p class="desc">Evolutionary Geneticist | <a href="https://www.ucl.ac.uk/" rel="external nofollow noopener" target="_blank">University College London</a></p>
        </header>

        <article>
          <div class="profile float-right">

              <figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/abhilesh_pic-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/abhilesh_pic-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/abhilesh_pic-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/assets/img/abhilesh_pic.png" class="img-fluid z-depth-1 rounded" width="auto" height="auto" alt="abhilesh_pic.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>

            <div class="address">
              <p>112, Darwin Building</p> <p>Gower Street</p> <p>London, UK - WC1E 6BT</p>

            </div>
          </div>

          <div class="clearfix">
              <p>I am an Evolutionary biologist passionate about understanding the mechanistic processes that produced the diversity of Life on Earth. I specialize in integrating empirical and computational approaches towards developing a holistic understanding of evolutionary processes.</p>

<p>Currently a Postdoctoral Research Fellow in the <a href="https://www.ucl.ac.uk/biosciences/gee" rel="external nofollow noopener" target="_blank">Genetics, Evolution and Environment</a> department at <a href="https://www.ucl.ac.uk/" rel="external nofollow noopener" target="_blank">University College London</a> with <a href="https://www.ucl.ac.uk/biosciences/people/max-reuter" rel="external nofollow noopener" target="_blank">Prof. Max Reuter</a> and <a href="https://www.ucl.ac.uk/biosciences/people/dr-aida-andres" rel="external nofollow noopener" target="_blank">Dr. Aida Andres</a>, my research focuses on how sex-specific traits evolve despite sharing a common genome. We use a combination of experimental evolution, genomics, and statistical modelling to understand the genetic basis of sexual antagonism.</p>

<p>Prior to this, I received my Ph.D. in Evolutionary Biology from the <a href="https://biosci.unl.edu/" rel="external nofollow noopener" target="_blank">School of Biological Sciences</a> at the <a href="https://www.unl.edu/" rel="external nofollow noopener" target="_blank">University of Nebraska-Lincoln</a>. I worked with <a href="https://biosci.unl.edu/kristi-montooth" rel="external nofollow noopener" target="_blank">Dr. Kristi Montooth</a> and <a href="https://biosci.unl.edu/colin-meiklejohn" rel="external nofollow noopener" target="_blank">Dr. Colin Meiklejohn</a> on the co-evolution between mitochondrial and nuclear genomes. Through phenotypic assays with <em>Drosophila</em> hybrids, I characterized the effects of incompatible mitochondrial-nuclear interactions on the thermal senstivity of male reproduction. Combining phylogenetic, biological sequence and protein structure analysis, I also characterized the characterized disease-causing mutations in proteins of the electron transport chain.</p>

<p>I recieved my Integrated BS-MS Dual Degree in Bioinformatics from the <a href="https://www.iiserpune.ac.in/" rel="external nofollow noopener" target="_blank">Indian Institute of Science Education and Research, Pune</a>, where I worked on research projects in Ecology and Evolutionary Biology</p>

          </div>

          <!-- News -->
          <h2><a href="/news/" style="color: inherit;">news</a></h2>          <div class="news">
            <div class="table-responsive">
              <table class="table table-sm table-borderless">
              
                <tr>
                  <th scope="row">Dec 22, 2023</th>
                  <td>
                    Selected for participation in the <a href="https://genetics-gsa.org/career-development/peer-review-training-program/" rel="external nofollow noopener" target="_blank">Genetics Society of America’s Peer Review Training Program</a>. I will be serving as a reviewer for the journals <em>Genetics</em> and <em>G3: Genes, Genomes, Genetics</em>.

                  </td>
                </tr>
                <tr>
                  <th scope="row">Jun 2, 2023</th>
                  <td>
                    Published a miniReview on <a href="https://www.sciencedirect.com/science/article/pii/S0959437X23000308" rel="external nofollow noopener" target="_blank">Multilevel Selection on mtDNA</a> in <em>Current Opinion in Genetics &amp; Development</em>

                  </td>
                </tr>
                <tr>
                  <th scope="row">Sep 5, 2022</th>
                  <td>
                    Joined <a href="https://www.ucl.ac.uk/biosciences/gee" rel="external nofollow noopener" target="_blank">University College London</a>’s GEE department as a Postdoctoral Research Fellow to work with Prof. Max Reuter and Dr. Aida Andres on Sexual Antagonistic Variation in <em>Drosophila melanogaster</em>.

                  </td>
                </tr>
              </table>
            </div>
          </div>

          <!-- Latest posts -->
          

          <!-- Selected papers -->
          <h2><a href="/publications/" style="color: inherit;">selected publications</a></h2>
          <div class="publications">
            <ol class="bibliography">
<li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/bioinfo2020-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/bioinfo2020-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/bioinfo2020-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/assets/img/publication_preview/bioinfo2020.jpeg" class="preview z-depth-1 rounded" width="auto" height="auto" alt="bioinfo2020.jpeg" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="dhawanjewarKnowledgebasedScoringFunction2020" class="col-sm-8">
        <!-- Title -->
        <div class="title">A Knowledge-Based Scoring Function to Assess Quaternary Associations of Proteins</div>
        <!-- Author -->
        <div class="author">
        

        <em>Abhilesh S Dhawanjewar</em>, <a href="https://scholar.google.com/citations?user=e8lZuhcAAAAJ&amp;hl=fr" rel="external nofollow noopener" target="_blank">Ankit A Roy</a>, and <a href="https://www.iiserpune.ac.in/research/department/biology/people/faculty/regular-faculty/m-s-madhusudhan/290" rel="external nofollow noopener" target="_blank">Mallur S Madhusudhan</a>
</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em>Bioinformatics</em>, Jun 2020
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
            <a href="https://academic.oup.com/bioinformatics/article/36/12/3739/5815973" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a>
            <a href="/assets/pdf/Dhawanjewar%20et%20al_2020_A%20knowledge-based%20scoring%20function%20to%20assess%20quaternary%20associations%20of%20proteins.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1093/bioinformatics/btaa207"></span>
              <span class="__dimensions_badge_embed__" data-doi="10.1093/bioinformatics/btaa207" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden abstract block -->
          <div class="abstract hidden">
            <p>Motivation The elucidation of all inter-protein interactions would significantly enhance our knowledge of cellular processes at a molecular level. Given the enormity of the problem, the expenses and limitations of experimental methods, it is imperative that this problem is tackled computationally. In silico predictions of protein interactions entail sampling different conformations of the purported complex and then scoring these to assess for interaction viability. In this study, we have devised a new scheme for scoring protein–protein interactions. Results Our method, PIZSA (Protein Interaction Z-Score Assessment), is a binary classification scheme for identification of native protein quaternary assemblies (binders/nonbinders) based on statistical potentials. The scoring scheme incorporates residue–residue contact preference on the interface with per residue-pair atomic contributions and accounts for clashes. PIZSA can accurately discriminate between native and non-native structural conformations from protein docking experiments and outperform other contact-based potential scoring functions. The method has been extensively benchmarked and is among the top 6 methods, outperforming 31 other statistical, physics based and machine learning scoring schemes. The PIZSA potentials can also distinguish crystallization artifacts from biological interactions. Availability and implementation PIZSA is implemented as a web server at http://cospi.iiserpune.ac.in/pizsa and can be downloaded as a standalone package from http://cospi.iiserpune.ac.in/pizsa/Download/Download.html. Supplementary information  are available at Bioinformatics online.</p>
          </div>
        </div>
      </div>
</li>
<li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/icbiol2019-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/icbiol2019-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/icbiol2019-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/assets/img/publication_preview/icbiol2019.jpeg" class="preview z-depth-1 rounded" width="auto" height="auto" alt="icbiol2019.jpeg" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="montoothTemperatureSensitiveReproductionPhysiological2019" class="col-sm-8">
        <!-- Title -->
        <div class="title">Temperature-Sensitive Reproduction and the Physiological and Evolutionary Potential for Mother’s Curse</div>
        <!-- Author -->
        <div class="author">
        

        <a href="https://biosci.unl.edu/kristi-montooth" rel="external nofollow noopener" target="_blank">Kristi L Montooth</a>, <em>Abhilesh S Dhawanjewar</em>, and <a href="https://biosci.unl.edu/colin-meiklejohn" rel="external nofollow noopener" target="_blank">Colin D Meiklejohn</a>
</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em>Integrative and Comparative Biology</em>, Jun 2019
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
            <a href="https://academic.oup.com/icb/article/59/4/890/5512493" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a>
            <a href="/assets/pdf/Montooth%20et%20al_2019_Temperature-Sensitive%20Reproduction%20and%20the%20Physiological%20and%20Evolutionary.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1093/icb/icz091"></span>
              <span class="__dimensions_badge_embed__" data-doi="10.1093/icb/icz091" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden abstract block -->
          <div class="abstract hidden">
            <p>\textcopyright 2019 The Author(s). Strict maternal transmission of mitochondrial DNA (mtDNA) is hypothesized to permit the accumulation of mitochondrial variants that are deleterious to males but not females, a phenomenon called mother’s curse. However, direct evidence that mtDNA mutations exhibit such sexually antagonistic fitness effects is sparse. Male-specific mutational effects can occur when the physiological requirements of the mitochondria differ between the sexes. Such male-specific effects could potentially occur if sex-specific cell types or tissues have energy requirements that are differentially impacted by mutations affecting energy metabolism. Here we summarize findings from a model mitochondrial-nuclear incompatibility in the fruit fly Drosophila that demonstrates sex-biased effects, but with deleterious effects that are generally larger in females. We present new results showing that the mitochondrial-nuclear incompatibility does negatively affect male fertility, but only when males are developed at high temperatures. The temperature-dependent male sterility can be partially rescued by diet, suggesting an energetic basis. Finally, we discuss fruitful paths forward in understanding the physiological scope for sex-specific effects of mitochondrial mutations in the context of the recent discovery that many aspects of metabolism are sexually dimorphic and downstream of sex-determination pathways in Drosophila. A key parameter of these models that remains to be quantified is the fraction of mitochondrial mutations with truly male-limited fitness effects across extrinsic and intrinsic environments. Given the energy demands of reproduction in females, only a small fraction of the mitochondrial mutational spectrum may have the potential to contribute to mother’s curse in natural populations.</p>
          </div>
        </div>
      </div>
</li>
<li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/nar2019-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/nar2019-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/nar2019-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/assets/img/publication_preview/nar2019.jpeg" class="preview z-depth-1 rounded" width="auto" height="auto" alt="nar2019.jpeg" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="royProteinInteractionScore2019" class="col-sm-8">
        <!-- Title -->
        <div class="title">Protein Interaction Z Score Assessment (PIZSA): An Empirical Scoring Scheme for Evaluation of Protein–Protein Interactions</div>
        <!-- Author -->
        <div class="author">
        

        <a href="https://scholar.google.com/citations?user=e8lZuhcAAAAJ&amp;hl=fr" rel="external nofollow noopener" target="_blank">Ankit A Roy</a>, <em>Abhilesh S Dhawanjewar</em>, <a href="https://parichitsharma.com/" rel="external nofollow noopener" target="_blank">Parichit Sharma</a>, and
          <span class="more-authors" title="click to view 2 more authors" onclick="
                var element = $(this);
                element.attr('title', '');
                var more_authors_text = element.text() == '2 more authors' ? 'Gulzar Singh, M S Madhusudhan' : '2 more authors';
                var cursorPosition = 0;
                var textAdder = setInterval(function(){
                  element.text(more_authors_text.substring(0, cursorPosition + 1));
                  if (++cursorPosition == more_authors_text.length){
                    clearInterval(textAdder);
                  }
              }, '10');
              ">2 more authors</span>
</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em>Nucleic Acids Research</em>, Jul 2019
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
            <a href="https://academic.oup.com/nar/article/47/W1/W331/5494716" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a>
            <a href="/assets/pdf/Roy%20et%20al_2019_Protein%20Interaction%20Z%20Score%20Assessment%20(PIZSA)" class="btn btn-sm z-depth-0" role="button">PDF</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1093/nar/gkz368"></span>
              <span class="__dimensions_badge_embed__" data-doi="10.1093/nar/gkz368" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden abstract block -->
          <div class="abstract hidden">
            <p>Our web server, PIZSA (http://cospi.iiserpune.ac.in/pizsa), assesses the likelihood of protein–protein interactions by assigning a Z Score computed from interface residue contacts. Our score takes into account the optimal number of atoms that mediate the interaction between pairs of residues and whether these contacts emanate from the main chain or side chain. We tested the score on 174 native interactions for which 100 decoys each were constructed using ZDOCK. The native structure scored better than any of the decoys in 146 cases and was able to rank within the 95th percentile in 162 cases. This easily outperforms a competing method, CIPS. We also benchmarked our scoring scheme on 15 targets from the CAPRI dataset and found that our method had results comparable to that of CIPS. Further, our method is able to analyse higher order protein complexes without the need to explicitly identify chains as receptors or ligands. The PIZSA server is easy to use and could be used to score any input three-dimensional structure and provide a residue pair-wise break up of the results. Attractively, our server offers a platform for users to upload their own potentials and could serve as an ideal testing ground for this class of scoring schemes.</p>
          </div>
        </div>
      </div>
</li>
</ol>
          </div>


          <!-- Social -->
        </article>

</div>

      
    </div>

    <!-- Footer -->    <footer class="sticky-bottom mt-5">
      <div class="container">
        © Copyright 2024 Abhilesh  Dhawanjewar. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>.

      </div>
    </footer>

    <!-- JavaScripts -->
    <!-- jQuery -->
  <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>

    <!-- Bootsrap & MDB scripts -->
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js" integrity="sha256-fgLAgv7fyCGopR/gBNq2iW3ZKIdqIcyshnUULC4vex8=" crossorigin="anonymous"></script>
  <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script>

    <!-- Masonry & imagesLoaded -->
  <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script>
  <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script>
  <script defer src="/assets/js/masonry.js" type="text/javascript"></script>
    
  <!-- Medium Zoom JS -->
  <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script>
  <script defer src="/assets/js/zoom.js"></script>

  <!-- Bootstrap Table -->
  <script defer src="https://unpkg.com/bootstrap-table@1.21.4/dist/bootstrap-table.min.js"></script>

  <!-- Load Common JS -->
  <script src="/assets/js/no_defer.js"></script>
  <script defer src="/assets/js/common.js"></script>
  <script defer src="/assets/js/copy_code.js" type="text/javascript"></script>

    
  <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
  <script async src="https://badge.dimensions.ai/badge.js"></script>

    <!-- MathJax -->
  <script type="text/javascript">
    window.MathJax = {
      tex: {
        tags: 'ams'
      }
    };
  </script>
  <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script>
  <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>

    
    

<!-- Scrolling Progress Bar -->
<script type="text/javascript">
  /*
   * This JavaScript code has been adapted from the article 
   * https://css-tricks.com/reading-position-indicator/ authored by Pankaj Parashar, 
   * published on the website https://css-tricks.com on the 7th of May, 2014.
   * Couple of changes were made to the original code to make it compatible 
   * with the `al-foio` theme.
   */
  const progressBar = $("#progress");
  /*
   * We set up the bar after all elements are done loading.
   * In some cases, if the images in the page are larger than the intended
   * size they'll have on the page, they'll be resized via CSS to accomodate
   * the desired size. This mistake, however, breaks the computations as the
   * scroll size is computed as soon as the elements finish loading.
   * To account for this, a minimal delay was introduced before computing the
   * values.
   */
  window.onload = function () {
    setTimeout(progressBarSetup, 50);
  };
  /*
   * We set up the bar according to the browser.
   * If the browser supports the progress element we use that.
   * Otherwise, we resize the bar thru CSS styling
   */
  function progressBarSetup() {
    if ("max" in document.createElement("progress")) {
      initializeProgressElement();
      $(document).on("scroll", function() {
        progressBar.attr({ value: getCurrentScrollPosition() });
      });
      $(window).on("resize", initializeProgressElement);
    } else {
      resizeProgressBar();
      $(document).on("scroll", resizeProgressBar);
      $(window).on("resize", resizeProgressBar);
    }
  }
  /*
   * The vertical scroll position is the same as the number of pixels that
   * are hidden from view above the scrollable area. Thus, a value > 0 is
   * how much the user has scrolled from the top
   */
  function getCurrentScrollPosition() {
    return $(window).scrollTop();
  }

  function initializeProgressElement() {
    let navbarHeight = $("#navbar").outerHeight(true);
    $("body").css({ "padding-top": navbarHeight });
    $("progress-container").css({ "padding-top": navbarHeight });
    progressBar.css({ top: navbarHeight });
    progressBar.attr({
      max: getDistanceToScroll(),
      value: getCurrentScrollPosition(),
    });
  }
  /*
   * The offset between the html document height and the browser viewport
   * height will be greater than zero if vertical scroll is possible.
   * This is the distance the user can scroll
   */
  function getDistanceToScroll() {
    return $(document).height() - $(window).height();
  }

  function resizeProgressBar() {
    progressBar.css({ width: getWidthPercentage() + "%" });
  }
  // The scroll ratio equals the percentage to resize the bar
  function getWidthPercentage() {
    return (getCurrentScrollPosition() / getDistanceToScroll()) * 100;
  }
</script>

  </body>
</html>
