<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta name="google-site-verification" content="_lH3zRueYRvwRbtYVH56oH7lgUs5HnrIL1SN0LAtKyY"> <meta http-equiv="Permissions-Policy" content="interest-cohort=()"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Publications | Abhilesh Dhawanjewar </title> <meta name="author" content="Abhilesh Dhawanjewar"> <meta name="description" content="More details about my publications can be found on my &lt;a href=https://scholar.google.com/citations?user=KmALIboAAAAJ&gt;Google Scholar profile&lt;/a&gt;."> <meta name="keywords" content="Bioinformatics, Mitochondrial Biology, Evolutionary Genetics"> <meta property="og:site_name" content="Abhilesh Dhawanjewar"> <meta property="og:type" content="website"> <meta property="og:title" content="Abhilesh Dhawanjewar | Publications"> <meta property="og:url" content="https://abhilesh.github.io/publications/"> <meta property="og:description" content="More details about my publications can be found on my &lt;a href=https://scholar.google.com/citations?user=KmALIboAAAAJ&gt;Google Scholar profile&lt;/a&gt;."> <meta property="og:locale" content="en"> <meta name="twitter:card" content="summary"> <meta name="twitter:title" content="Publications"> <meta name="twitter:description" content="More details about my publications can be found on my &lt;a href=https://scholar.google.com/citations?user=KmALIboAAAAJ&gt;Google Scholar profile&lt;/a&gt;."> <script type="application/ld+json">
    

    {
      "@context": "https://schema.org",
      "@type": "WebSite",
      "url": "https://abhilesh.github.io/publications/",
      "name": "Abhilesh Dhawanjewar",
      
        "headline": "Abhilesh Dhawanjewar | Evolutionary Geneticist \u0026 Bioinformatician",
      

      "author": {
        "@type": "Person",
        "name": "Abhilesh Dhawanjewar",
        "url": "https://abhilesh.github.io"
        ,

        
        "jobTitle": "Bioinformatician"
        ,
        

        
        "affiliation": {
          "@type": "Organization",
          "name": "MRC Mitochondrial Biology Unit, University of Cambridge"
          
          ,"url": "https://www.mrc-mbu.cam.ac.uk/"
          
        }

        ,
        

        
          "description": "Dr. Abhilesh Dhawanjewar is an Evolutionary Geneticist and Bioinformatician at the MRC Mitochondrial Biology Unit, University of Cambridge. He specializes in the evolutionary genetics of mitochondrial DNA and its implications for human health and disease."
          ,
        

        
          "sameAs": ["https://orcid.org/0000-0001-7827-1255","https://scholar.google.com/citations?user=KmALIboAAAAJ","https://www.researchgate.net/profile/Abhilesh-Dhawanjewar-2","https://github.com/abhilesh","https://www.linkedin.com/in/abhilesh-dhawanjewar","https://twitter.com/abhilesh7","https://neuroscience.cam.ac.uk/member/ad2347/","https://www.mrc-mbu.cam.ac.uk/mbu-postdoctoral-society/postdocs/abhilesh-dhawanjewar","https://theealingbeaverproject.com/team/abhilesh-dhawanjewar/","https://instagram.com/abhilesh7"]
        ,
        

        
        "image": {
            "@type": "ImageObject",
            "url": "https://abhilesh.github.io/assets/img/abhilesh_pic.png",
            "height": "200",
            "width": "200"
        }
        
      }

      
    }
    
  </script> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="/assets/libs/mdb/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" href="/assets/css/scholar-icons.css?62b2ac103a88034e6882a5be5f3e2772"> <link defer rel="stylesheet" type="text/css" href="/assets/libs/google_fonts/google-fonts.css"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="/assets/img/favicon-32x32.png?61747260cb6c82387f9d334dad0b5cd9"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://abhilesh.github.io/publications/"> <script src="/assets/js/theme.js?9a0c749ec5240d9cda97bc72359a72c0"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-monokai.css?d536d0f87ce647fcf6b5eff9db01df15" media="none" id="highlight_theme_dark"> <script>
    initTheme();
  </script> </head> <body class="fixed-top-nav sticky-bottom-footer"> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> Abhilesh <span class="font-weight-bold">Dhawanjewar </span> </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">About </a> </li> <li class="nav-item active"> <a class="nav-link" href="/publications/">Publications <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">Blog </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">CV </a> </li> <li class="nav-item"> <button id="search-toggle" title="Search" onclick="openSearchModal()"> <span class="nav-link">ctrl k <i class="ti ti-search"></i></span> </button> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main" style="--stagger: 0;" data-animate> <div class="post"> <header class="post-header"> <h1 class="post-title">Publications</h1> <p class="post-description" style="--stagger: 1;" data-animate="">More details about my publications can be found on my <a href="https://scholar.google.com/citations?user=KmALIboAAAAJ" rel="external nofollow noopener" target="_blank">Google Scholar profile</a>.</p> </header> <article> <script src="/assets/js/bibsearch.js?1bc438ca9037884cc579601c09afd847" type="module"></script> <p style="--stagger: 2;" data-animate=""><input type="text" id="bibsearch" spellcheck="false" autocomplete="off" class="search bibsearch-form-input" placeholder="Type to filter"></p> <div class="publications"> <h2 class="bibliography" style="--stagger: 3;" data-animate="">2023</h2> <ol class="bibliography"><li> <div class="row" style="--stagger: 4;" data-animate=""> <div class="col-sm-2 preview"> <figure style="--stagger: 5;" data-animate=""> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/cogde2023-480.webp 480w,/assets/img/publication_preview/cogde2023-800.webp 800w,/assets/img/publication_preview/cogde2023-1400.webp 1400w," type="image/webp" sizes="50px"></source> <img src="/assets/img/publication_preview/cogde2023.jpeg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="cogde2023.jpeg" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="camusMultilevelSelectionMitochondrial2023" class="col-sm-8"> <div class="title">Multilevel Selection on Mitochondrial Genomes</div> <div class="author"> <a href="https://www.ucl.ac.uk/biosciences/people/dr-florencia-camus" rel="external nofollow noopener" target="_blank">M. Florencia Camus</a>, and <em>Abhilesh S Dhawanjewar</em> </div> <div class="periodical"> <em>Current Opinion in Genetics &amp; Development</em>, Jun 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.sciencedirect.com/science/article/pii/S0959437X23000308" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/Camus_Dhawanjewar_2023_Multilevel%20selection%20on%20mitochondrial%20genomes.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p style="--stagger: 6;" data-animate="">Mitochondria are vital organelles for life in eukaryotes, taking centre stage in the process of cellular respiration. This process is regulated via a series of finely coordinated obligate interactions of molecules encoded by two genomes: nuclear DNA and mitochondrial DNA. Both genomes are required to work harmoniously to provide cellular energy, with detrimental consequences occurring when there is miscommunication between them. Whilst the need for cooperation is strong, vast differences between genomes (ploidy, size, and inheritance) create an arena for conflict. Here, we examine the varying levels of selection operating on the mitochondrial genome and the consequences they have on all these levels. We conclude by highlighting the potential for conflict when selection at different levels is driven by different evolutionary forces.</p> </div> </div> </div> </li></ol> <h2 class="bibliography" style="--stagger: 7;" data-animate="">2022</h2> <ol class="bibliography"><li> <div class="row" style="--stagger: 8;" data-animate=""> <div class="col-sm-2 abbr"> <abbr class="badge">PhD Thesis</abbr> </div> <div id="dhawanjewarTaleTwoGenomes2022" class="col-sm-8"> <div class="title">A Tale of Two Genomes: The Complex Interplay between the Mitochondrial and the Nuclear Genomes</div> <div class="author"> <em>Abhilesh S Dhawanjewar</em> </div> <div class="periodical"> <em>The University of Nebraska-Lincoln</em>, Jun 2022 </div> <div class="periodical"> </div> <div class="links"> <a href="/assets/pdf/Dhawanjewar%20-%20A%20Tale%20of%20Two%20Genomes%20The%20Complex%20Interplay%20Betwe.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> </div> </div> </li></ol> <h2 class="bibliography" style="--stagger: 9;" data-animate="">2020</h2> <ol class="bibliography"><li> <div class="row" style="--stagger: 10;" data-animate=""> <div class="col-sm-2 preview"> <figure style="--stagger: 11;" data-animate=""> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/bioinfo2020-480.webp 480w,/assets/img/publication_preview/bioinfo2020-800.webp 800w,/assets/img/publication_preview/bioinfo2020-1400.webp 1400w," type="image/webp" sizes="50px"></source> <img src="/assets/img/publication_preview/bioinfo2020.jpeg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="bioinfo2020.jpeg" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="dhawanjewarKnowledgebasedScoringFunction2020" class="col-sm-8"> <div class="title">A Knowledge-Based Scoring Function to Assess Quaternary Associations of Proteins</div> <div class="author"> <em>Abhilesh S Dhawanjewar</em>, <a href="https://scholar.google.com/citations?user=e8lZuhcAAAAJ&amp;hl=fr" rel="external nofollow noopener" target="_blank">Ankit A Roy</a>, and <a href="https://www.iiserpune.ac.in/research/department/biology/people/faculty/regular-faculty/m-s-madhusudhan/290" rel="external nofollow noopener" target="_blank">Mallur S Madhusudhan</a> </div> <div class="periodical"> <em>Bioinformatics</em>, Jun 2020 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://academic.oup.com/bioinformatics/article/36/12/3739/5815973" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/Dhawanjewar%20et%20al_2020_A%20knowledge-based%20scoring%20function%20to%20assess%20quaternary%20associations%20of%20proteins.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p style="--stagger: 12;" data-animate="">Motivation The elucidation of all inter-protein interactions would significantly enhance our knowledge of cellular processes at a molecular level. Given the enormity of the problem, the expenses and limitations of experimental methods, it is imperative that this problem is tackled computationally. In silico predictions of protein interactions entail sampling different conformations of the purported complex and then scoring these to assess for interaction viability. In this study, we have devised a new scheme for scoring protein–protein interactions. Results Our method, PIZSA (Protein Interaction Z-Score Assessment), is a binary classification scheme for identification of native protein quaternary assemblies (binders/nonbinders) based on statistical potentials. The scoring scheme incorporates residue–residue contact preference on the interface with per residue-pair atomic contributions and accounts for clashes. PIZSA can accurately discriminate between native and non-native structural conformations from protein docking experiments and outperform other contact-based potential scoring functions. The method has been extensively benchmarked and is among the top 6 methods, outperforming 31 other statistical, physics based and machine learning scoring schemes. The PIZSA potentials can also distinguish crystallization artifacts from biological interactions. Availability and implementation PIZSA is implemented as a web server at http://cospi.iiserpune.ac.in/pizsa and can be downloaded as a standalone package from http://cospi.iiserpune.ac.in/pizsa/Download/Download.html. Supplementary information are available at Bioinformatics online.</p> </div> </div> </div> </li></ol> <h2 class="bibliography" style="--stagger: 13;" data-animate="">2019</h2> <ol class="bibliography"> <li> <div class="row" style="--stagger: 14;" data-animate=""> <div class="col-sm-2 preview"> <figure style="--stagger: 15;" data-animate=""> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/icbiol2019-480.webp 480w,/assets/img/publication_preview/icbiol2019-800.webp 800w,/assets/img/publication_preview/icbiol2019-1400.webp 1400w," type="image/webp" sizes="50px"></source> <img src="/assets/img/publication_preview/icbiol2019.jpeg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="icbiol2019.jpeg" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="montoothTemperatureSensitiveReproductionPhysiological2019" class="col-sm-8"> <div class="title">Temperature-Sensitive Reproduction and the Physiological and Evolutionary Potential for Mother’s Curse</div> <div class="author"> <a href="https://biosci.unl.edu/kristi-montooth" rel="external nofollow noopener" target="_blank">Kristi L Montooth</a>, <em>Abhilesh S Dhawanjewar</em>, and <a href="https://biosci.unl.edu/colin-meiklejohn" rel="external nofollow noopener" target="_blank">Colin D Meiklejohn</a> </div> <div class="periodical"> <em>Integrative and Comparative Biology</em>, Jun 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://academic.oup.com/icb/article/59/4/890/5512493" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/Montooth%20et%20al_2019_Temperature-Sensitive%20Reproduction%20and%20the%20Physiological%20and%20Evolutionary.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p style="--stagger: 16;" data-animate="">Strict maternal transmission of mitochondrial DNA (mtDNA) is hypothesized to permit the accumulation of mitochondrial variants that are deleterious to males but not females, a phenomenon called mother’s curse. However, direct evidence that mtDNA mutations exhibit such sexually antagonistic fitness effects is sparse. Male-specific mutational effects can occur when the physiological requirements of the mitochondria differ between the sexes. Such male-specific effects could potentially occur if sex-specific cell types or tissues have energy requirements that are differentially impacted by mutations affecting energy metabolism. Here we summarize findings from a model mitochondrial-nuclear incompatibility in the fruit fly Drosophila that demonstrates sex-biased effects, but with deleterious effects that are generally larger in females. We present new results showing that the mitochondrial-nuclear incompatibility does negatively affect male fertility, but only when males are developed at high temperatures. The temperature-dependent male sterility can be partially rescued by diet, suggesting an energetic basis. Finally, we discuss fruitful paths forward in understanding the physiological scope for sex-specific effects of mitochondrial mutations in the context of the recent discovery that many aspects of metabolism are sexually dimorphic and downstream of sex-determination pathways in Drosophila. A key parameter of these models that remains to be quantified is the fraction of mitochondrial mutations with truly male-limited fitness effects across extrinsic and intrinsic environments. Given the energy demands of reproduction in females, only a small fraction of the mitochondrial mutational spectrum may have the potential to contribute to mother’s curse in natural populations.</p> </div> </div> </div> </li> <li> <div class="row" style="--stagger: 17;" data-animate=""> <div class="col-sm-2 preview"> <figure style="--stagger: 18;" data-animate=""> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/molecol2019-480.webp 480w,/assets/img/publication_preview/molecol2019-800.webp 800w,/assets/img/publication_preview/molecol2019-1400.webp 1400w," type="image/webp" sizes="50px"></source> <img src="/assets/img/publication_preview/molecol2019.jpeg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="molecol2019.jpeg" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="nelsonUNVEILingConnectionsGenotype2019" class="col-sm-8"> <div class="title">UNVEILing Connections between Genotype, Phenotype, and Fitness in Natural Populations</div> <div class="author"> <a href="https://www.thomnelson.com/" rel="external nofollow noopener" target="_blank">Thomas C Nelson</a>, <a href="https://scholar.google.com/citations?user=XtTUlvIAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Matthew R Jones</a>, <a href="https://science.du.edu/about/faculty-directory/jonathan-velotta" rel="external nofollow noopener" target="_blank">Jonathan P Velotta</a>, and <span class="more-authors" title="click to view 2 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '2 more authors' ? 'Abhilesh S Dhawanjewar, Rena M Schweizer' : '2 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">2 more authors</span> </div> <div class="periodical"> <em>Molecular Ecology</em>, Jun 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://onlinelibrary.wiley.com/doi/full/10.1111/mec.15067" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/Nelson%20et%20al_2019_UNVEILing%20connections%20between%20genotype,%20phenotype,%20and%20fitness%20in%20natural" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p style="--stagger: 19;" data-animate="">Understanding the links between genetic variation and fitness in natural populations is a central goal of evolutionary genetics. This monumental task spans the fields of classical and molecular genetics, population genetics, biochemistry, physiology, developmental biology, and ecology. Advances to our molecular and developmental toolkits are facilitating integrative approaches across these traditionally separate fields, providing a more complete picture of the genotype-phenotype map in natural and non-model systems. Here, we summarize research presented at the first annual symposium of the UNVEIL Network, an NSF-funded collaboration between the University of Montana and the University of Nebraska, Lincoln, which took place from the 1st to the 3rd of June, 2018. We discuss how this body of work advances basic evolutionary science, what it implies for our ability to predict evolutionary change, and how it might inform novel conservation strategies.</p> </div> </div> </div> </li> <li> <div class="row" style="--stagger: 20;" data-animate=""> <div class="col-sm-2 preview"> <figure style="--stagger: 21;" data-animate=""> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/nar2019-480.webp 480w,/assets/img/publication_preview/nar2019-800.webp 800w,/assets/img/publication_preview/nar2019-1400.webp 1400w," type="image/webp" sizes="50px"></source> <img src="/assets/img/publication_preview/nar2019.jpeg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="nar2019.jpeg" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="royProteinInteractionScore2019" class="col-sm-8"> <div class="title">Protein Interaction Z Score Assessment (PIZSA): An Empirical Scoring Scheme for Evaluation of Protein–Protein Interactions</div> <div class="author"> <a href="https://scholar.google.com/citations?user=e8lZuhcAAAAJ&amp;hl=fr" rel="external nofollow noopener" target="_blank">Ankit A Roy</a>, <em>Abhilesh S Dhawanjewar</em>, <a href="https://parichitsharma.com/" rel="external nofollow noopener" target="_blank">Parichit Sharma</a>, and <span class="more-authors" title="click to view 2 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '2 more authors' ? 'Gulzar Singh, M S Madhusudhan' : '2 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">2 more authors</span> </div> <div class="periodical"> <em>Nucleic Acids Research</em>, Jul 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://academic.oup.com/nar/article/47/W1/W331/5494716" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/Roy%20et%20al_2019_Protein%20Interaction%20Z%20Score%20Assessment%20(PIZSA)" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p style="--stagger: 22;" data-animate="">Our web server, PIZSA (http://cospi.iiserpune.ac.in/pizsa), assesses the likelihood of protein–protein interactions by assigning a Z Score computed from interface residue contacts. Our score takes into account the optimal number of atoms that mediate the interaction between pairs of residues and whether these contacts emanate from the main chain or side chain. We tested the score on 174 native interactions for which 100 decoys each were constructed using ZDOCK. The native structure scored better than any of the decoys in 146 cases and was able to rank within the 95th percentile in 162 cases. This easily outperforms a competing method, CIPS. We also benchmarked our scoring scheme on 15 targets from the CAPRI dataset and found that our method had results comparable to that of CIPS. Further, our method is able to analyse higher order protein complexes without the need to explicitly identify chains as receptors or ligands. The PIZSA server is easy to use and could be used to score any input three-dimensional structure and provide a residue pair-wise break up of the results. Attractively, our server offers a platform for users to upload their own potentials and could serve as an ideal testing ground for this class of scoring schemes.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography" style="--stagger: 23;" data-animate="">2015</h2> <ol class="bibliography"><li> <div class="row" style="--stagger: 24;" data-animate=""> <div class="col-sm-2 abbr"> <abbr class="badge">MS Thesis</abbr> </div> <div id="dhawanjewar_statistical_2015" class="col-sm-8"> <div class="title">Statistical Potentials for Prediction of Protein-Protein Interactions</div> <div class="author"> <em>Abhilesh S Dhawanjewar</em> </div> <div class="periodical"> <em>Indian Institute of Science Education and Research, Pune</em>, Jul 2015 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="/assets/pdf/Dhawanjewar%20-%202015%20-%20Statistical%20Potentials%20for%20Prediction%20of%20Protein-Protein%20Interactions.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p style="--stagger: 25;" data-animate="">Protein-Protein Interactions are critical to life, playing crucial roles in a variety of cellular processes. Hence, prediction of protein-protein interactions would help in gaining insights into cellular processes so that we may be able to manipulate and control it. In this study, we have developed knowledge-based pairwise statistical potentials based on experimentally derived structures for the prediction of protein-protein complexes. Structures of protein dimers in the Protein Data Bank (PDB) were used for the construction of the statistical potentials. A total of 96 different pairwise potentials were constructed for different values of five parameters: distance threshold for interactions, interacting atom types, weight type, weighting scheme and reference state. The performance of these potentials was benchmarked using Receiver Operating Characteristics (ROC) curves and Rank-Ordering. The side chain-side chain pairwise potentials were the best performers keeping all other parameters constant. The best performing pairwise potential could discriminate native structures from a sequence-randomized background in a benchmark set of 296 structures with a false positive rate of 1.4% and a true positive rate of 98.6%. This result is an improvement over the MODTIE potential which had a false positive rate of 28.5% and a true positive rate of 71.5%. The pairwise potentials are also complementary to each other, in the sense that they are efficient on different subsets of the benchmark set. Hence, a combination of the different potentials could result in better prediction accuracy. An attempt towards the development of a 5-body potential based on the pairwise potential was also initiated. Two different versions, an unweighted and a weighted potential were developed. The weighted multi-body potentials performed better than the unweighted potential. These multi-body potentials will be further refined, which is a work in progress. This prediction system will be bundled into a web server in the near future.</p> </div> </div> </div> </li></ol> </div> </article> </div> </div> <footer class="sticky-bottom mt-5" role="contentinfo"> <div class="container"> © Copyright 2025 Abhilesh Dhawanjewar. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. </div> </footer> <script src="/assets/libs/jquery/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="/assets/libs/mdb/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="/assets/libs/masonry/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="/assets/libs/imagesloaded/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js?a0db7e5d5c70cc3252b3138b0c91dcaf" type="text/javascript"></script> <script defer src="/assets/libs/medium_zoom/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?00311c0cb6a8fd1f19211a0bf42ce53d"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script defer type="text/javascript" id="MathJax-script" src="/assets/libs/mathjax/tex-mml-chtml.js" integrity="sha256-qF6qU5kTY5jf4hgs0vrVP18n4V2AyF/Xw3jOo0K6yCU=" crossorigin="anonymous"></script> <script src="/assets/js/mathjax-setup.js?a5bb4e6a542c546dd929b24b8b236dfd"></script> <script defer src="/assets/libs/polyfill/polyfill.min.js" crossorigin="anonymous"></script> <script async src="https://www.googletagmanager.com/gtag/js?id=G-M1BWYLLR80"></script> <script>
    window.dataLayer = window.dataLayer || [];
    function gtag() {
      dataLayer.push(arguments);
    }
    gtag('js', new Date());

    gtag('config', 'G-M1BWYLLR80');
  </script> <script defer src="/assets/js/google-analytics-setup.js?12374742c4b1801ba82226e617af7e2d"></script> <script defer src="/assets/js/progress-bar.js?2f30e0e6801ea8f5036fa66e1ab0a71a" type="text/javascript"></script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>
    addBackToTop();
  </script> <script type="module" src="/assets/js/search/ninja-keys.min.js?a3446f084dcaecc5f75aa1757d087dcf"></script> <ninja-keys hidebreadcrumbs noautoloadmdicons placeholder="Type to start searching"></ninja-keys> <script src="/assets/js/search-setup.js?6c304f7b1992d4b60f7a07956e52f04a"></script> <script src="/assets/js/search-data.js"></script> <script src="/assets/js/shortcut-key.js?6f508d74becd347268a7f822bca7309d"></script> </body> </html>